recipes/cvplot/all/conanfile.py

```diff
@@ -0,0 +1,36 @@
+from conans import ConanFile, tools
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class CvPlotConan(ConanFile):
+    name = "cvplot"
+    description = "fast modular opencv plotting library"
+    license = "MIT"
+    topics = ("plot", "opencv", "diagram", "plotting")
+    homepage = "https://github.com/Profactor/cv-plot"
+    url = "https://github.com/conan-io/conan-center-index"
+    generators = "cmake"
+    requires = "opencv/4.5.3"
+    no_copy_source = True
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version], strip_root=True, destination=self._source_subfolder)
+
+    def package(self):
+        self.copy(pattern="LICENSE", dst="licenses", src=self._source_subfolder)
+        self.copy(pattern="*", dst="include", src=os.path.join(self._source_subfolder, "CvPlot", "inc"))
+        
+    def package_id(self):
+        self.info.header_only()
+            
+    def package_info(self):
+        self.cpp_info.defines.append("CVPLOT_HEADER_ONLY")
+        self.cpp_info.set_property("cmake_find_mode", "both") 
```

Comment:
I was not able to find any modules being installed 🤔 

```suggestion
```

Reply:
Not sure about that. I thought `both` means, that we support both `find_package(CvPlot CONFIG)` and `find_package(CvPlot MODULE)`?

I tried the `MODULE` version. Conan creates a `Findcvplot.cmake` then but cmake does not find opencv:

```
-- Found cvplot: 1.2.2 (found version "1.2.2")
-- Conan: Target declared 'cvplot::cvplot'
CMake Error at /usr/share/cmake-3.16/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  No "Findopencv.cmake" found in CMAKE_MODULE_PATH.
Call Stack (most recent call first):
  build/7e3f3c9f8b0319ca5c01315759aeda2f877da74f/FindCvPlot.cmake:25 (find_dependency)
  CMakeLists.txt:7 (find_package) 
```

So I guess we should stick to the default `config` mode?

Reply:
have to add

```
self.cpp_info.set_property("cmake_file_name", "CvPlot")
self.cpp_info.set_property("cmake_target_name", "CvPlot::CvPlot")
```

Reply:
That solves the `CvPlot` spelling, but I still dont know how to consume the `FindCvPlot.cmake`. Always getting `No "Findopencv.cmake"`.

Reply:
hmm, Conan generates OpenCV.cmake not opencv.cmake. Do you have a log? The CI is working for the test package.

Reply:
The current test_package works of course, but I wanted to see if `set_property("cmake_find_mode", "both")` does something useful. From the documentation, `cmake_find_mode` is used for `CMakeDeps`. So I modified
- `test_package/CMakeLists.txt` line 7 to `find_package(CvPlot MODULE REQUIRED)`
- `test_package/conanfile.py` line 7 to `generators = "cmake", "CMakeDeps"`
Cmake output:

```
-- Found CvPlot: 1.2.2 (found version "1.2.2")
-- Conan: Target declared 'CvPlot::CvPlot'
CMake Error at /usr/share/cmake-3.16/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):
  No "Findopencv.cmake" found in CMAKE_MODULE_PATH.
Call Stack (most recent call first):
  build/7e3f3c9f8b0319ca5c01315759aeda2f877da74f/FindCvPlot.cmake:25 (find_dependency)
  CMakeLists.txt:7 (find_package)
```

It works with `generators = "cmake", "cmake_find_package"` which also creates `FindOpenCV.cmake`, but that does not require `set_property("cmake_find_mode", "both")`.

Reply:
> that we support both

Who is we? 🤔 

My remark is in regards to the upstream project which searching https://github.com/Profactor/cv-plot/search?l=CMake&q=install does not seem to install any CMake targets or config files.

This is a new setting but typically do not change the settings unless it's require by the upstream project.

Our status quo, as I recall (I've been less active lately), is consumer a project via a "non official" means requires using the default Conan settings (e.g. omitting this settings).

***

As for your question about using the `find_package(... MODULE)` to the best of my knowledge Modules can only find other modules not `CONFIG` packages.
I am not sure if that was every improved on inn recent CMake version but the [docs](https://cmake.org/cmake/help/latest/command/find_package.html#id4) dont give me hope. However it could be a Conan bug 😱

I would assume the OpenCV recipe has the default `CONFIG` mode set and looking for a module that depends on a config is not something I would expect to work.

Reply:
We == you and me😊
I agree, so we remove `self.cpp_info.set_property("cmake_find_mode", "both")`. Ok @uilianries ?

Reply:
I think some consumers may use `find_package(cvplot CONFIG)` and some may use `find_package(cvplot MODULE)`, right? is there any reason not to support both use-cases?

Reply:
@SSE4 as I said above:
- `find_package(cvplot MODULE)` already works with `generators = "cmake", "cmake_find_package"` and does not need `self.cpp_info.set_property("cmake_find_mode", "both")`
- The only problem is with `generators = "cmake", "CMakeDeps"` (`No "Findopencv.cmake" found`)

Reply:
According to the CMake docs https://cmake.org/cmake/help/latest/manual/cmake-modules.7.html#find-modules`find_package(cvplot MODULE)` does not exist ... to my knowledge. I do not expect it to work

Reply:
this list built-in CMake modules only, but someone can use the custom Find module in his own project, right? I've seen it in a lot of OSS projects in the wild, it's pretty common practice

Reply:
Sadly theres been a lot of misuse... downstream project can write their own (which is fairly common) then this setting makes sense

https://cmake.org/cmake/help/latest/command/find_package.html#id4

The file CMake searches for in module mode should not be provided by the package itself.

***

Looking in the github repo I dont see any custom files... but it's not easy to search all projects